internal/pkg/cli/pipeline_init.go

```diff
@@ -158,19 +157,57 @@ func (o *initPipelineOpts) Validate() error {
 		return err
 	}
 
-	if o.repoURL != "" {
-		if err := o.validateURL(o.repoURL); err != nil {
-			return err
-		}
-	}
-
 	if o.environments != nil {
 		for _, env := range o.environments {
 			_, err := o.store.GetEnvironment(o.appName, env)
 			if err != nil {
 				return err
 			}
 		}
+	}
+
+	// If both '--url' and '--git-branch' flags are passed in.
+	if o.repoURL != "" && o.repoBranch != "" {
+		// Validate if URL of accepted source type (GitHub, Bitbucket, CodeCommit).
+		if err := o.validateURLType(o.repoURL); err != nil {
+			return err
+		}
+		// Validate if URL is an existent git remote and store the remote name.
+		if err := o.validateURLExists(o.repoURL); err != nil {
+			return err
+		}
+		// Validate that branch exists for the given git remote.
+		if err := o.validateBranch(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// If only the '--url' flag is passed in, ask for branch.
+	if o.repoURL != "" {
+		// Validate if URL of accepted source type (GitHub, Bitbucket, CodeCommit).
+		if err := o.validateURLType(o.repoURL); err != nil {
+			return err
+		}
+
+		var err error
+		o.repoRemoteName, err = o.getRemoteNameFromURL()
+		if err != nil {
+			return err
+		}
+		if err := o.selectBranch(); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	// If only the '--git-branch' is passed in, ask for URL.
+	if o.repoBranch != "" {
+		// Ask user to select URL if it is not set by flag.
+		// Remote repo choices include only those that have the passed in branch.
+		if err := o.filterThenSelectReposByBranch(); err != nil {
```

Comment:
Okay. I misunderstood your comment https://github.com/aws/copilot-cli/pull/3182#discussion_r784286346. Will revert.

Reply:
The reason I had originally made it so that users had to pass in both flags is that in order to validate branch, we need URL. In the past, we didn't validate that the passed-in URL or branch exists, but I think we all agreed it would be better to validate them before the pipeline building proceeded.

Reply:
I think it might be fine if they only provide one flag and we don't validate that far. So if they only pass the URL flag then we check if the repo exists, whereas if they only pass the branch flag we can't really validate it until in `Ask()` when they pass in the URL. Eventually we'll error out if there's no available option for them.

For example, if they pass in `master` for the branch flag and then in `Ask()` we get to know their URL. However, when finding available options for them the length is 0 then we'll error out (if it is 1 we'll make it default instead of asking for them to choose)